lib/metasploit/framework/login_scanner/softing_sis.rb

```diff
@@ -79,9 +73,35 @@ def do_login(user, pass)
             return { status: LOGIN_STATUS::INCORRECT, proof: auth_res.body.to_s }
           end
 
+          auth_token
+        end
+
+        # generate a signature from the authentication token, username, and password
+        #
+        # @param auth_token [String] The authentication token retrieved by calling get_auth_token
+        # @param user [String] The username
+        # @param pass [String] The password
+        # @return [String] A hexadecimal string representation of the signature
+        def generate_signature(auth_token, user, pass)
+          Digest::MD5.hexdigest(auth_token + pass + auth_token + user + auth_token)
+        end
+
+        # the actual login method, called by #attempt_login
+        #
+        # @param user [String] The username to try
+        # @param pass [String] The password to try
+        # @return [Hash]
+        #   * status [Metasploit::Model::Login::Status]
+        #   * proof [String] the HTTP response body
+        def do_login(user, pass)
+          # prep the data needed for login
+          protocol = ssl ? 'https' : 'http'
+          # attempt to get an authentication token
+          auth_token = get_auth_token(user)
```

Comment:
The return value from `auth_token` here needs to be checked before proceeding because the `#get_auth_token` method can fail for a few different reasons.

Reply:
Agreed - I made it so that `#do_login` returns early with the hash from `#get_auth_token` if the value of `status` is not `LOGIN_STATUS::SUCCESSFUL`.